debezium-testing/debezium-testcontainers-testing/src/test/java/io/debezium/testing/testcontainers/DebeziumContainerTest.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.testing.testcontainers;
+
+import static org.fest.assertions.Assertions.assertThat;
+
+import java.io.IOException;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.testcontainers.containers.KafkaContainer;
+import org.testcontainers.containers.PostgreSQLContainer;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+public class DebeziumContainerTest {
+
+    private static PostgreSQLContainer postgreContainer = new PostgreSQLContainer<>("debezium/postgres:11");
+    private static KafkaContainer kafkaContainer = new KafkaContainer("5.3.1");
+    private static DebeziumContainer debeziumContainer = new DebeziumContainer("1.0.0.Final");
+
+    @BeforeClass
+    public static void initializeInfrastructure() {
+        kafkaContainer.start();
+
+        postgreContainer.withNetwork(kafkaContainer.getNetwork()).start();
```

Comment:
consider explicitly setting the network on KafkaContainer, otherwise it will print a deprecation warning.

Also, you can define a network at the configuration time, and then use `Startables.deepStart(Stream.of(kafkaContainer, postgreContainer, debeziumContainer)).join()` to start everything in parallel (you can also use `dependsOn` to have some dependencies)

Reply:
Do you have a complete example for `Startables#deepStart()`? Where do I need to put this, some `@BeforeClass` method?

Reply:
`Startables#deepStart()` is a helper to recursively start containers in parallel with their dependencies, so that your tests start faster.

You can use it in place of custom `kafkaContainer.start()`-like calls.

To say it differently - remove everything from `initializeInfrastructure`, replace with a single `deepStart` and make sure you configure the network (you can use `withNetwork(Network.SHARED)` in case you don't need a strict network isolation)

Reply:
@bsideup, ok, so this means I wouldn't use the `@Rule` annotation then?

Reply:
yes, same as in the current code, just the containers will be started in parallel.

Think about `Startables#deepStart()` as something similar to `Stream.of(kafka, postgres, debezium).parallel().forEach(it -> it.start())` but also with deep traversal and a dedicated thread pool

Reply:
Ok, largely rewrote the test. Really like it now. Thanks a lot for your guidance!